pkg/server/datastore/sqlstore/sqlstore_test.go

```diff
@@ -3877,225 +3868,59 @@ func (s *PluginSuite) TestUpdateFederationRelationship() {
+		s.T().Run(fmt.Sprintf("migration_from_schema_version_%d", schemaVersion), func(t *testing.T) {
+			require := require.New(t)
+			dbName := fmt.Sprintf("v%d.sqlite3", schemaVersion)
+			dbPath := filepath.ToSlash(filepath.Join(s.dir, "migration-"+dbName))
+			dbURI := fmt.Sprintf("file://%s", dbPath)
+
+			minimalDB := func() string {
+				previousMinor := codeVersion
+				if codeVersion.Minor == 0 {
+					previousMinor.Major--
+				} else {
+					previousMinor.Minor--
+				}
+				return fmt.Sprintf(`
+					CREATE TABLE "migrations" ("id" integer primary key autoincrement, "version" integer,"code_version" varchar(255) );
+					INSERT INTO migrations("version", "code_version") VALUES (%d,%q);
+				`, schemaVersion, previousMinor)
+			}
 
-			// Assert current serial numbers and expiration time remains the same
-			expectedTime, err := time.Parse(time.RFC3339, "2018-12-19T15:26:58-07:00")
-			s.Require().NoError(err)
-			s.Require().Equal(expectedTime.Unix(), attestedNode.CertNotAfter)
-			s.Require().Equal("111", attestedNode.CertSerialNumber)
-
-			// Assert the new fields are empty for pre-existing entries
-			s.Require().Empty(attestedNode.NewCertSerialNumber)
-			s.Require().Empty(attestedNode.NewCertNotAfter)
-		case 13:
-			s.Require().True(s.ds.db.Dialect().HasColumn("registered_entries", "revision_number"))
-		case 14:
-			db, err := openSQLite3(dbURI)
-			defer db.Close()
-			s.Require().NoError(err)
-			s.Require().True(db.Dialect().HasIndex("attested_node_entries", "idx_attested_node_entries_expires_at"))
-		case 15:
-			s.Require().True(s.ds.db.Dialect().HasColumn("registered_entries", "store_svid"))
-		case 16:
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "trust_domain"))
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "bundle_endpoint_url"))
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "bundle_endpoint_profile"))
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "endpoint_spiffe_id"))
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "implicit"))
-			s.Require().True(s.ds.db.Dialect().HasIndex("federated_trust_domains", "uix_federated_trust_domains_trust_domain"))
-		case 17:
-			s.Require().True(s.ds.db.Dialect().HasColumn("attested_node_entries", "can_reattest"))
-			s.Require().True(s.ds.db.Dialect().HasColumn("registered_entries", "hint"))
-		default:
-			s.T().Fatalf("no migration test added for version %d", i)
-		}
+			prepareDB := func(migrationSupported bool) {
+				dump := migrationDumps[schemaVersion]
+				if migrationSupported {
+					require.NotEmpty(dump, "no migration dump set up for schema version")
+				} else {
+					require.Empty(dump, "migration dump exists for unsupported schema version")
+					dump = minimalDB()
+				}
+				dumpDB(t, dbPath, dump)
+				err := s.ds.Configure(fmt.Sprintf(`
+				database_type = "sqlite3"
+				connection_string = %q
+			`, dbURI))
```

Comment:
nit: indentation looks a bit off here

Reply:
Oh yeah. Woof! Other spots I've indented one more than this.